### PR TITLE
Wallet integration into starter app

### PR DESCRIPTION
**Pull request by elv-seth**

### CODE REVIEW

**Review by elv-wayne — COMMENTED** [review]
Can you check if `npm run lint` gives no errors? I see a pre-commit key in the package.json but not sure if that's working.  Other than some indentation issues it's a good start.

**Comment by elv-kevin** [comment]
I would prefer that we don't use / link to elv-components-js

Also, I'd recommend removing pre-commit

**Comment by elv-wayne** [comment]
> 

We're using the media-apps branch of elv-components

**Comment by elv-zenia** [comment]
@elv-kevin Have you run into issues with pre-commits?

**Comment by elv-seth** [comment]
Just pushed a new commit addressing the issues I found with "npm run lint" and some formatting.

File: src/components/GettingStarted.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+div {
+    background-color: rgba(13,13,13,1);
+    width: 100%;
+}
+
+p {
+    text-align: center;
+    vertical-align: middle;
+    color: white;
```
Comment (elv-zenia) [inline src/components/GettingStarted.css]: Always use color variables in css. Check out /elv-components-js/src/stylesheets/variables/colors.scss.

File: src/components/GettingStarted.js [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+  box-shadow: 0px 0px 12px 0px rgba(97, 51, 255, 1);
+};
+&:disabled {
+  opacity: 0.7;
+  background-color: rgba(104,51,255,1);
+}
+margin: auto;
+`;
+
+
```
Comment (elv-zenia) [inline src/components/GettingStarted.js]: Move styling to a stylesheet. Also look into replacing PrimaryButton with an `Action` component from elv-components-js

File: src/components/Home.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      <div style = {{height: "100px"}}/>
+      <p>Hello {name.split(" ")[0]},</p>
+      <p>This project template uses and demonstrates the following Eluvio projects:</p>
+      <a href = "https://github.com/eluv-io/elv-media-wallet" target="_blank"><p style =  {{color: "rgba(81,184,240,1)"}}>-elv-media-wallet</p></a>
+      <a href = "https://github.com/eluv-io/elv-components-js" target="_blank"><p style = {{color: "rgba(81,184,240,1)"}}>-elv-components-js</p></a>
+      <div style = {{height: "100px"}}/>
+      <PrimaryButton onClick = {
+        Logout
+      }><p>Logout</p></PrimaryButton>
+    </div>
```
Comment (elv-zenia) [inline src/components/Home.js]: Use React fragments https://reactjs.org/docs/fragments.html
```suggestion
    <>
      <div style = {{height: "100px"}}/>
      <p>Hello {name.split(" ")[0]},</p>
      <p>This project template uses and demonstrates the following Eluvio projects:</p>
      <a href = "https://github.com/eluv-io/elv-media-wallet" target="_blank"><p style =  {{color: "rgba(81,184,240,1)"}}>-elv-media-wallet</p></a>
      <a href = "https://github.com/eluv-io/elv-components-js" target="_blank"><p style = {{color: "rgba(81,184,240,1)"}}>-elv-components-js</p></a>
      <div style = {{height: "100px"}}/>
      <PrimaryButton onClick = {
        Logout
      }><p>Logout</p></PrimaryButton>
    </>
```

File: src/components/Home.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    `;
+    
+
+
+  return (
+    <div>
+      <div style = {{height: "100px"}}/>
+      <p>Hello {name.split(" ")[0]},</p>
+      <p>This project template uses and demonstrates the following Eluvio projects:</p>
+      <a href = "https://github.com/eluv-io/elv-media-wallet" target="_blank"><p style =  {{color: "rgba(81,184,240,1)"}}>-elv-media-wallet</p></a>
```
Comment (elv-zenia) [inline src/components/Home.js]: Replace anchor tags with elv-components-js `<Link>`

File: src/router/index.js [hunk 5]
```diff
@@ -1,19 +1,62 @@
...
+import {Routes, Route} from "react-router-dom";
 
 import GettingStarted from "Components/GettingStarted";
 import Components from "Components/Components";
-import Fabric from "Components/Fabric";
+import Home from "Components/Home";
+import Navbar from "Components/Navbar";
+import { ElvWalletClient } from "@eluvio/elv-wallet-client"; 
+
+
```
Comment (elv-zenia) [inline src/router/index.js:11]: Remove extra lines. Check throughout your changes. There generally shouldn't be more than one consecutive empty line 
Will also look into adding a linting rule for this

File: src/router/index.js [hunk 6]
```diff
@@ -1,19 +1,62 @@
...
+
+const AppRoutes = React.memo(() => {
+
+  const [profile, setProfile] = useState({});
+  const [loggedIn,setLoggedIn] = useState(false);
```
Comment (elv-zenia) [inline src/router/index.js]: Can you move `loggedIn` state to a data store? We don't have any set up yet but the structure should be src/stores/index.js (rootStore). Look at elv-music-js's rootStore and elv-media-wallet's rootStore

**Review by elv-zenia — CHANGES_REQUESTED** [review]
This is awesome, Seth. A few notes in addition to the comments I left. 

- When setting properties, don't use spaces in between equal signs and brackets, e.g., `src={}`.

- Stylesheets should go in src/static/stylesheets and should be .scss, not .css.
- Don't use inline styling. Keep it separate from JS in stylesheets.
- Be careful with extra lines. Use no more than one empty line if separating code blocks.
- You shouldn't need styled-components if using elv-components.
Lastly, can you try to implement a navbar from scratch instead of using Material?

File: package.json [hunk 7]
```diff
@@ -22,12 +22,18 @@
...
+    "flux": "^4.0.3",
+    "material-ui": "^0.20.2",
     "mobx": "^6.6.0",
     "mobx-react": "^7.5.0",
     "react": "^18.1.0",
     "react-dom": "^18.1.0",
-    "react-router-dom": "^6.3.0"
+    "react-icons": "^4.4.0",
+    "react-router-dom": "^6.3.0",
+    "styled-components": "^5.3.5"
```
Comment (elv-zenia) [inline package.json]: Can you remove @material-ui/core, material-ui, react-icons, and styled-components?

File: src/components/GettingStarted.js [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
+import {Action} from "elv-components-js";
 
-const GettingStarted = () => {
+const GettingStarted = ({Login}) => {
   return (
-    <div>Getting Started</div>
+    <div >
+      <img src={logo} />
+      <div className="spacer">
+        <Action label="login"title="Login"type="button"className="primarybtn"onClick={Login}children={<p className="btntext">Login</p>} />
```
Comment (elv-zenia) [inline src/components/GettingStarted.js]: Sorry, I think I added confusion about the spaces
```suggestion
        <Action label="login" title="Login" type="button" className="primarybtn" onClick={Login} children={<p className="btntext">Login</p>} />
```

File: src/components/Navbar.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Link} from "react-router-dom";
+
+const Navbar = () => {
+  return (
+    <nav className="nav">
+      <ul>
+        <Link to="/" >Home</Link>
```
Comment (elv-zenia) [inline src/components/Navbar.js]:
```suggestion
        <Link to="/">Home</Link>
```

File: src/index.js [hunk 10]
```diff
@@ -2,17 +2,16 @@ import React from "react";
...
 import Routes from "./router";
 import {HashRouter} from "react-router-dom";
-
-import NavigationBar from "Components/NavigationBar";
 import "Assets/stylesheets/app.scss";
 
 const rootElement = ReactDOM.createRoot(document.getElementById("app"));
 
+
+
```
Comment (elv-zenia) [inline src/index.js]: Extra lines

**Review by elv-zenia — CHANGES_REQUESTED** [review]
Left comments for minor things.
I see that you're using Flux, which I believe is integrated into Redux since I think they follow the paradigm of needing to dispatch values in order to update state. The difference with Mobx--and one of the advantages-- is that it serves as a data layer that automatically changes the UI layer to reflect any changes in state. So the idea is to set up the store so that it contains observables and actions, which mutate state. 

Like so
```
class RootStore {
  loggedIn = false;

  constructor() {
    makeAutoObservable(this);
  }

  Authenticate() {
    this.loggedIn = true;
  }
}
```

And then set up your component so that it imports the store and is an observer.

```
import {rootStore} from "Stores/index";

const SomeComponent = observer (() => {
  return (
    <div>{ rootStore.loggedIn }</div>
  );
});
```
So in the simple component above, the value of `loggedIn` will change every time the state changes in the store.
The Mobx ten-minute intro is a solid read on this: https://mobx.js.org/getting-started

And, of course, let me know if you want to talk about this more.

**Review by elv-zenia — APPROVED** [review]

**Pull request merged**